<launch>
  <!-- Set args -->
  <arg name="robot_name"/>
  <arg name="size" value="0.16" />
  <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="node_namespace" default="apriltag_ros_continuous_node" />

  <!-- Set parameters -->
  <rosparam command="load" file="$(find apriltag_ros)/config/settings.yaml" ns="$(arg node_namespace)" />
  
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="camera/image_raw" />
    <remap from="camera_info" to="camera/camera_info" />
    <param name="publish_tag_detections_image" type="bool" value="false" />      <!-- default: false -->
    <rosparam param="standalone_tags" subst_value="True">[
      {id: 0, size: $(arg size), name: "tag_0/$(arg robot_name)"},
      {id: 1, size: $(arg size), name: "tag_1/$(arg robot_name)"},
      {id: 2, size: $(arg size), name: "tag_2/$(arg robot_name)"},
      {id: 3, size: $(arg size), name: "tag_3/$(arg robot_name)"},
      {id: 4, size: $(arg size), name: "tag_4/$(arg robot_name)"},
      {id: 5, size: $(arg size), name: "tag_5/$(arg robot_name)"},
      {id: 6, size: $(arg size), name: "tag_6/$(arg robot_name)"},
      {id: 7, size: $(arg size), name: "tag_7/$(arg robot_name)"},
      {id: 8, size: $(arg size), name: "tag_8/$(arg robot_name)"},
      {id: 9, size: $(arg size), name: "tag_9/$(arg robot_name)"},
      {id: 10, size: $(arg size), name: "tag_10/$(arg robot_name)"},
      {id: 11, size: $(arg size), name: "tag_11/$(arg robot_name)"},
      {id: 12, size: $(arg size), name: "tag_12/$(arg robot_name)"},
      {id: 13, size: $(arg size), name: "tag_13/$(arg robot_name)"}]
    </rosparam>
  </node>
</launch>
